event.go

```diff
@@ -107,21 +117,26 @@ func Eventf(sev Severity, ctx context.Context, msg string, params ...interface{}
 		Timestamp: time.Now().UTC(),
 		Severity:  sev,
 		Message:   msg,
-		Metadata:  metadata}
+		Metadata:  metadata,
+	}
 }
 
-type logMetadataProvider interface {
-	LogMetadata() map[string]string
+func stringMapToInterfaceMap(m map[string]string) map[string]interface{} {
+	shim := map[string]interface{}{}
```

Comment:
This would be slightly more performant if you did this, but it's not a big deal.

```suggestion
	shim := make(map[string]interface{}, len(m))
```